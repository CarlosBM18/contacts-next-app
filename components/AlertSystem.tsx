import { observer } from "mobx-react-lite";
import React, { useCallback, useEffect } from "react";
import { AlertElementProps, AlertObject } from "../libs/types";
import { useStore } from "../stores";
import styles from "../styles/components/AlertSystem.module.css";

export const AlertSystem = observer(() => {
  const { rootStore } = useStore();

  return (
    <div className={styles.alertSystemContainer}>
      {rootStore.alertsStore.alerts.map((alert, index) => (
        <Alert data={alert} key={index} />
      ))}
    </div>
  );
});

const Alert = ({ data }: AlertElementProps) => {
  const { rootStore } = useStore();

  const removeAlert = useCallback(() => {
    rootStore.alertsStore.removeAlert(data.id);
  }, [data.id, rootStore.alertsStore]);

  useEffect(() => {
    setTimeout(() => {
      removeAlert();
    }, 6000);
  }, [data.id, removeAlert, rootStore.alertsStore]);

  const getAlertStyle = () => {
    switch (data.status) {
      case "success":
        return styles.alertSuccess;
      case "error":
        return styles.alertError;
    }
  };

  return (
    <div
      className={`${styles.alertContainer} ${getAlertStyle()}`}
      onClick={removeAlert}
    >
      {data.message}
    </div>
  );
};
