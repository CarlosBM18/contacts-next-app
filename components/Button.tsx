import React from "react";
import { ButtonProps } from "../libs/types";
import styles from "../styles/components/Button.module.css";

export const Button = ({ title, onClick, type }: ButtonProps) => {
  function buttonType() {
    switch (type) {
      case "primary":
        return styles.button;
      case "danger":
        return `${styles.button} ${styles.buttonDanger}`;
      default:
        return styles.button;
    }
  }
  return (
    <div className={buttonType()} onClick={onClick}>
      {title}
    </div>
  );
};
