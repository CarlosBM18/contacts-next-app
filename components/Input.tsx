import React from "react";
import { InputProps } from "../libs/types";
import styles from "../styles/components/Input.module.css";

export const Input = ({ label, value, setValue, type }: InputProps) => {
  return (
    <div className={styles.content}>
      <div className={styles.label}>{label}</div>
      <input
        className={styles.input}
        value={value}
        type={type}
        onChange={(event: any) => {
          setValue(event.target.value);
        }}
      />
    </div>
  );
};
