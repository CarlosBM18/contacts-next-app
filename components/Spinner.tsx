import React from "react";
import styles from "../styles/components/Spinner.module.css";

export const Spinner = () => {
  return <div className={styles.spinner}></div>;
};
