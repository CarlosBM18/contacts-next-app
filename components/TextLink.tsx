import React from "react";
import { TextLinkProps } from "../libs/types";
import styles from "../styles/components/TextLink.module.css";

export const TextLink = ({ text, onClick }: TextLinkProps) => {
  return (
    <div className={styles.textLink} onClick={onClick}>
      {text}
    </div>
  );
};
