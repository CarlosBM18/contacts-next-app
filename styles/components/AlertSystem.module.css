.alertSystemContainer {
  position: absolute;
  right: 2rem;
  top: 10vh;
}

.alertContainer {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  width: 250px;
  cursor: pointer;
  transition: right 0.6s;
  height: auto;
  right: -300px;
}

.alertContainer.show {
  right: 0px;
}

.alertContainer::first-letter {
  text-transform: uppercase;
}

.alertSuccess {
  color: #155724;
  background-color: #d4edda;
  border-color: #c3e6cb;
}

.alertError {
  color: #721c24;
  background-color: #f8d7da;
  border-color: #f5c6cb;
}
