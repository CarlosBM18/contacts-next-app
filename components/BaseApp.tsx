import React, { useEffect } from "react";
import { AppProps } from "next/dist/shared/lib/router/router";
import { observer } from "mobx-react-lite";
import { useStore } from "../stores";
import { useRouter } from "next/router";
import { isAuthRoute } from "../libs/utils";
import { Spinner } from "./Spinner";

export const BaseApp = observer(({ Component, pageProps }: AppProps) => {
  const { rootStore } = useStore();
  const router = useRouter();

  // Redirect user to login if not logged
  useEffect(() => {
    if (!rootStore.appStore.isLogged && !isAuthRoute(router.pathname)) {
      router.replace("/login");
    }
  }, [
    rootStore.appStore.isLogged,
    rootStore.appStore.loadingUser,
    rootStore.appStore.token,
    router,
  ]);

  // Redirect user if tries to go to auth routes when logged
  useEffect(() => {
    if (isAuthRoute(router.pathname) && rootStore.appStore.isLogged) {
      router.replace("/");
    }
  }, [
    rootStore.appStore.isLogged,
    rootStore.appStore.loadingUser,
    rootStore.appStore.token?.length,
    router,
    router.pathname,
  ]);

  // Return spinner when loading
  if (
    (rootStore.appStore.loadingUser || !rootStore.appStore.isLogged) &&
    !isAuthRoute(router.pathname)
  ) {
    return <Spinner />;
  }

  return <Component {...pageProps} />;
});
