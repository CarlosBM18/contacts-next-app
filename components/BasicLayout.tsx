import { observer } from "mobx-react-lite";
import Head from "next/head";
import { BasicLayoutProps } from "../libs/types";
import { useStore } from "../stores";
import styles from "../styles/components/BasicLayout.module.css";
import { TextLink } from "./TextLink";

const BasicLayout = observer(({ title, children }: BasicLayoutProps) => {
  const { rootStore } = useStore();

  const onLogout = () => {
    rootStore.appStore.saveToken("", localStorage);
  };

  return (
    <div className={styles.container}>
      <Head>
        <title>{title}</title>
      </Head>

      <main className={styles.main}>
        <div className={styles.titleContainer}>
          <h1 className={styles.title}>{title}</h1>
        </div>
        <div className={styles.content}>{children}</div>
      </main>

      <footer className={styles.footer}>
        <a>Creaded by CarlosBM</a>
        {!!rootStore.appStore.token?.length && (
          <div className={styles.logoutContainer}>
            <TextLink text="Logout" onClick={onLogout} />
          </div>
        )}
      </footer>
    </div>
  );
});

export { BasicLayout };
