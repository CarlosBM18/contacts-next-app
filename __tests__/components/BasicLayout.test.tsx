import React from "react";
import renderer from "react-test-renderer";
import { BasicLayout } from "../../components/BasicLayout";

test("renders BasicLayout correctly", () => {
  const tree = renderer
    .create(
      <BasicLayout title="Test">
        <div>Test</div>
      </BasicLayout>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
