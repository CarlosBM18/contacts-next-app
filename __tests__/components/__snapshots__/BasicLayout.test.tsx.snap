// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BasicLayout correctly 1`] = `
<div
  className="container"
>
  <main
    className="main"
  >
    <div
      className="titleContainer"
    >
      <h1
        className="title"
      >
        Test
      </h1>
    </div>
    <div
      className="content"
    >
      <div>
        Test
      </div>
    </div>
  </main>
  <footer
    className="footer"
  >
    <a>
      Creaded by CarlosBM
    </a>
  </footer>
</div>
`;
